### PR TITLE
Reset live counters on sandbox change

### PR DESCRIPTION
**Pull request by CompuIves**
We had some side effects that were still running when we were forking a sandbox, this resets them.

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3745/commits/7e7471e8a5a315f12f8aee70c2d0745cb33cb254 is at https://pr3745.build.csb.dev/s/new.

File: packages/app/src/app/overmind/effects/live/index.ts [hunk 1]
```diff
@@ -495,6 +495,18 @@ class Live {
     });
   }
 
+  reset() {
+    this.clients.clear();
+
+    if (this.awaitSendTimer) {
+      clearTimeout(this.awaitSendTimer);
+    }
```
Comment (christianalfoni) [inline packages/app/src/app/overmind/effects/live/index.ts]:
```suggestion
    clearTimeout(this.awaitSendTimer);
```

No need to check :-)

File: packages/app/src/app/overmind/effects/live/index.ts [hunk 2]
```diff
@@ -495,6 +495,18 @@ class Live {
...
+
+    if (this.awaitSend) {
+      this.awaitSend = null;
+    }
```
Comment (christianalfoni) [inline packages/app/src/app/overmind/effects/live/index.ts]:
```suggestion
    this.awaitSend = null;
```

No need to check, just set it to null :-)

File: packages/app/src/app/overmind/namespaces/editor/internalActions.ts [hunk 3]
```diff
@@ -72,7 +72,7 @@ export const initializeLiveSandbox: AsyncAction<Sandbox> = async (
       // to join
       state.live.joinSource === 'sandbox'
     ) {
-      await actions.live.internal.disconnect();
```
Comment (christianalfoni) [inline packages/app/src/app/overmind/namespaces/editor/internalActions.ts]: Sure you do not want to wait until the disconnect is done?

Comment (CompuIves) [reply packages/app/src/app/overmind/namespaces/editor/internalActions.ts]: The function doesn't return a promise, so this is a bit misleading.

**Comment by CompuIves** [comment]
Got an 👍 on Slack!

**Pull request merged**